### DISCUSSION TITLE
Create3 Room Prevention

### DISCUSSION THREAD

**Discussion by SeguraUT** [iRobotEducation/create3_docs#348] [Questions]
Hello everyone, our group is looking to do some research using the creata3 Roomba. We want these to move around a hallway but are not allowed to have them go into private rooms (bathrooms, offices). Is there a hardware or software option to prevent this? Any ideas would greatly help!

Thank you

**Comment by shamlian** [comment]
The iRobot Create 3 educational robot is a primarily a sensing and mobility platform; while it has fused odometry, you likely would need to supply your own sensors and compute to do SLAM well enough to exclude it from entering particular areas. The only "creative" solution that might work could be to engineer barriers that could trigger one of the robot's hazard reflexes.

**Reply by SeguraUT** [reply]
We were thinking about putting QR codes near doorways and placing some type of vision sensor on top so it would ignore that area.
